packages/remix-dev/config.ts

```diff
@@ -260,21 +260,21 @@ export async function readConfig(
   remixRoot?: string,
   serverMode = ServerMode.Production
 ): Promise<RemixConfig> {
-  if (!remixRoot) {
-    remixRoot = process.env.REMIX_ROOT || process.cwd();
-  }
-
   if (!isValidServerMode(serverMode)) {
     throw new Error(`Invalid server mode "${serverMode}"`);
   }
 
+  if (!remixRoot) {
+    remixRoot = process.env.REMIX_ROOT || process.cwd();
+  }
+
```

Comment:
@ftonato - out of curiosity, what's the benefit of moving this after the `if (!isValidServerMode(serverMode)) {` conditional block?

Reply:
@cliffordfajardo you should consider that if you have code like this:

``` js
const fn = x => {
    if (!x) throw new Error('Invalid parameter')
    
    let a = x // assign something to a variable
    console.log(a)
}

fn() /* if we don't have the parameter it will throw the error
        before "assigning" something that might not be used for anything */
```